src/transformers/models/bloom/modeling_flax_bloom.py

```diff
@@ -0,0 +1,771 @@
+                variable_axes={"params": 0, "cache": 0},
+                split_rngs={"params": True, "dropout": True},
+                in_axes=(nn.broadcast, nn.broadcast, 0, nn.broadcast, nn.broadcast, nn.broadcast),
+                length=self.config.num_hidden_layers,
+            )(self.config, dtype=self.dtype, use_scan=True, name="FlaxBloomBlockLayers")(
+                hidden_states,
+                alibi,
+                attention_mask,  # kwargs not supported by scan
+                jnp.arange(self.config.num_hidden_layers),
+                None,
+                deterministic,
+                init_cache,
+            )
+            hidden_states = hidden_states[0]
+
+        else:
+            for layer_number in range(self.config.num_hidden_layers):
+                if output_hidden_states:
+                    all_hidden_states += (hidden_states,)
+
+                layer_outputs = FlaxBloomBlock(self.config, name=str(layer_number), dtype=self.dtype, use_scan=False)(
+                    hidden_states,
+                    alibi=alibi,
+                    attention_mask=attention_mask,
+                    deterministic=deterministic,
+                    init_cache=init_cache,
+                    output_attentions=output_attentions,
+                    layer_number=layer_number,
+                )
+                hidden_states = layer_outputs[0]
+
+                if output_attentions:
+                    all_attentions += (layer_outputs[1],)
+
+        # this contains possible `None` values - `FlaxBloomModule` will filter them out
+        outputs = (hidden_states, all_hidden_states, all_attentions)
+
+        return outputs
+
+
+class FlaxBloomModule(nn.Module):
+    config: BloomConfig
+    dtype: jnp.dtype = jnp.float32
+    use_scan: bool = False
+
+    def setup(self):
+        self.embed_dim = self.config.hidden_size
+
+        embedding_init = jax.nn.initializers.normal(stddev=self.config.initializer_range)
+
+        # word embeddings (no positional embedding layer)
+        self.word_embeddings = nn.Embed(
+            self.config.vocab_size,
+            self.embed_dim,
+            embedding_init=embedding_init,
+            dtype=self.dtype,
+        )
+
+        # post-embedding layernorm
+        self.word_embeddings_layernorm = nn.LayerNorm(epsilon=self.config.layer_norm_epsilon, dtype=self.dtype)
+
+        # transformer layers
+        self.h = FlaxBloomBlockCollection(self.config, dtype=self.dtype, use_scan=self.use_scan)
+
+        # final layernorm
+        self.ln_f = nn.LayerNorm(epsilon=self.config.layer_norm_epsilon, dtype=self.dtype)
+
+    def __call__(
+        self,
+        input_ids=None,
+        attention_mask=None,
+        deterministic=True,
+        init_cache: bool = False,
+        output_attentions: bool = False,
+        output_hidden_states: bool = False,
+        return_dict: bool = True,
+    ):
+        inputs_embeds = self.word_embeddings(input_ids.astype("i4"))
```

Comment:
`input_ids` are already casted to `i4` in the pre-trained model class no ?

Reply:
Some tests where failing when removing the casting, probably the `input_ids` are not casted to `i4` in the pre-trained model class - will investigate that

Reply:
Thanks for pointing out this issue! Finally fixed in 05c05212fb629487227b5e20b16a57995b3709c7